### ISSUE TITLE
router-helper provider configured $stateChangeError does not fire

### ISSUE THREAD

**Issue by ruionwriting** [johnpapa/angular-styleguide#402]
When data services $http results (catch promise) errors. I've spend hours around this and I have no idea what is going on. must certain I'm not seeing this on the right perspective.

So my plan is to grab all resource response errors on a global fashion way so I'm considering probably a good idea would be to to add an  interceptor to $httpProvider.

**Comment by johnpapa** [comment]
im confused. what is the issue? the title says state change error while you are talking about http and interceptors.

**Comment by Markionium** [comment]
I'd assume he means when using a Promise in route `resolve` which does some `$http` stuff that fails the event is not triggered.

**Comment by ruionwriting** [comment]
That is correct @Markionium.

**Comment by johnpapa** [comment]
perhaps the catch is not returning the rejected promise

**Comment by ruionwriting** [comment]
This is what I have on the catch:

```JavaScript
return $q.reject(msg);
```

**Comment by wardbell** [comment]
Sounds like a bug in your code. Would like to see more.

I doubt an `$httpProvider` interceptor would help much. Yes if the failure was in the http processing but that's what your outside catch is for. The interceptor won't help at all for mistakes that occur after the respond arrives ... and I'll bet your frustrations are traced to that phase of the promise chain.

**Comment by johnpapa** [comment]
closing
